src/crypto/hex_base.h

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2009-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_CRYPTO_HEX_BASE_H
+#define BITCOIN_CRYPTO_HEX_BASE_H
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/28690#discussion_r1417977573

> Is this commit needed? I guess there could be issues if the same translation unit is compiled twice?
> 
> Not sure if Hex is correct to put into "crypto".
> 
> If this isn't needed, maybe leave this for a follow-up?
> 
> An alternative would be to just remove it from the util library and keep it only in consensus?

Not sure about the current state, but when this was asked previously https://github.com/bitcoin/bitcoin/pull/28690#pullrequestreview-1734365185, I think the idea was to have it in crypto since util and consensus libraries depend on the crypto library.

Reply:
If you drop the commit splitting `strencodings` and moving the hex function to the crypto library you get a bunch of errors when compiling the last commit:
```
/usr/bin/ld: /usr/bin/ld: DWARF error: invalid or unhandled FORM value: 0x23
util/.libs/libbitcoinkernel_la-strencodings.o: in function `SanitizeString[abi:cxx11](std::basic_string_view<char, std::char_traits<char> >, int)':
strencodings.cpp:(.text+0x0): multiple definition of `SanitizeString[abi:cxx11](std::basic_string_view<char, std::char_traits<char> >, int)'; /usr/bin/ld: DWARF error: invalid or unhandled FORM value: 0x23
```
This is because the external `libbitcoinkernel` library now re-uses the existing source file lists and will find two `strencodings` files, one from the util sources and one from the consensus sources.